txn.go

```diff
@@ -683,15 +717,36 @@ func (txn *Txn) Get(table, index string, args ...interface{}) (ResultIterator, e
 
 	// Create an iterator
 	iter := &radixIterator{
+		done:    txn.trackIterator(),
 		iter:    indexIter,
 		watchCh: watchCh,
 	}
 	return iter, nil
 }
 
+func (txn *Txn) trackIterator() func() {
+	txn.activeIterators++
+	var called bool
+	return func() {
+		if !called {
+			called = true
+			txn.activeIterators--
+		}
+	}
+}
+
+func (txn *Txn) checkActiveIterator() error {
+	if txn.activeIterators > 0 {
+		return fmt.Errorf("operation is not safe from within a ResultIteration.Next loop")
```

Comment:
I think this error message can be improved, but I'll wait to see what others thing about this option.